### DISCUSSION TITLE
Frequent network requests to `http://169.254.169.254/metadata/instance/compute`

### DISCUSSION THREAD

**Discussion by database64128** [community/community#137229] [Copilot Conversations]
### Select Topic Area

Bug

### Body

Every few seconds, the copilot extension attempts to query `http://169.254.169.254/metadata/instance/compute`, and then opens a connection to `westus-0.in.applicationinsights.azure.com:443`. What are the purposes of these requests? Why does the extension send them at such high frequency?

- Copilot Chat Extension Version: v0.18.2
- VS Code Version: 1.92.2
- OS Version: Arch Linux, macOS, Windows

ref microsoft/vscode-copilot-release#1496

**Comment by suryanarayan-nadar** [comment]
Demo

**Comment by SukkaW** [comment]
<img width="1846" alt="image" src="https://github.com/user-attachments/assets/8ea1565f-cfec-4651-828a-cff2c3bfa3bb">

I am also facing the same issue as shown above. I have manually configured the firewall to drop `169.254.169.254` requests.

**Comment by Mindset-Art-Collective** [comment]
I'm seeing similar issues from Copilot Coding Agent on GitHub (not local VS Code). Coding sessions error out after all work is complete.

Copilot stopped work on behalf of [-redcated-](https://github.com/-redcated-) due to an error [17 hours ago](https://github.com/-redcated-/-redacted-/pull/393#event-19761650559)
[Copilot has encountered an error. See logs for additional details.](https://github.com/redcated/redcated/pull/393/agent-sessions/fc5fa261-ceb8-4d72-85d5-7d301045ca0c)

Details from the session logs indicate: 


⚠️ Warning: I tried to connect to the following addresses, but was blocked by firewall rules:


If you need me to access, download, or install something from one of these locations, you can either:

  - Configure Actions setup steps (https://gh.io/copilot/actions-setup-steps) to set up my env before the firewall is enabled.
  - Add the appropriate URLs or hosts to my firewall allow list (https://gh.io/copilot/firewall-config)


{"time":"2025-09-17T23:40:20.874197178Z","level":"WARN","msg":"PACKET BLOCKED","because":"IPNotAllowed","blocked":true,"blockedAt":"packet","domains":"None","ruleSource":"Unknown","ruleSourceComment":"Unknown","url":"","cmd":"/usr/bin/python3 -u bin/WALinuxAgent-2.14.0.1-py3.12.egg -collect-logs ","originalIp":"169.254.169.254","ip":"127.0.0.1","port":"34669","pid":3833,"hasBeenRedirected":true,"eventType":"Normal Packet"}

**Comment by ghost** [comment]
still hasnt been fixed https://github.com/microsoft/vscode-copilot-release/issues/1496 yet @lramos15 locked the issue

**Reply by lramos15** [reply]
I still am waiting on a fix from the upstream package, and I didn't lock the issue. The repository the issue was in got archived which locks all issues there. That was not intentional. Feel free to file a new issue in microsoft/vscode

**Comment by tylerharris82836-netizen** [comment]
That sounds like a telemetry or environment-detection loop, especially with the Application Insights connection; I’d first check whether it changes with telemetry disabled or across extension versions, alongside any related [subscription management tools](https://lemsubs.com/tools).
